internal/provisioner/plain/validator/validator.go

```diff
@@ -0,0 +1,60 @@
+package validator
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	apimacherrors "k8s.io/apimachinery/pkg/util/errors"
+	apimachyaml "k8s.io/apimachinery/pkg/util/yaml"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func Validate(bundle fs.FS) apimacherrors.Aggregate {
+	var validationErrors []error
+	objects, err := GetObjects(bundle)
+	if err != nil {
+		validationErrors = append(validationErrors, fmt.Errorf("get objects from bundle manifests: %v", err))
+	} else if len(objects) == 0 {
+		validationErrors = append(validationErrors, errors.New("invalid bundle: found zero objects: "+
+			"plain+v0 bundles are required to contain at least one object"))
+	}
+	return apimacherrors.NewAggregate(validationErrors)
+}
+
+func GetObjects(bundleFS fs.FS) ([]pkgclient.Object, error) {
```

Comment:
Seems a bit odd to me that `GetObjects` is in the `validate` package. Perhaps we should have a `internal/bundle/plain/` package that contains a `LoadBundleV0(fs.FS) (*BundleV0, error)` function.

And then all of the logic for enforcing the plain+v0 bundle spec would be contained there (and could still be abstracted out into a separate set of validations functions)?